runtime/v2/store.go

```diff
@@ -55,3 +57,32 @@ type Store interface {
 	// LastCommitID returns the latest commit ID
 	LastCommitID() (proof.CommitID, error)
 }
+
+// StoreLoader allows for custom loading of the store, this is useful when upgrading the store from a previous version
+type StoreLoader func(store Store) error
+
+// DefaultStoreLoader just calls LoadLatestVersion on the store
+func DefaultStoreLoader(store Store) error {
+	return store.LoadLatestVersion()
+}
+
+func V2UpgradeStoreLoader(upgradeHeight int64, storeUpgrades *store.StoreUpgrades) StoreLoader {
```

Comment:
can we add a go doc?

Reply:
Given that it is in runtime/v2, no need to prepend it by `V2` imho.
For consistency, shall we place the other one from x/upgrade in runtime?
And just alias it back in x/upgrade for not introducing breaking changes?

Additionally, can we add a `:tip:` in x/upgrade docs mentioning for server/v2 this helper is in runtime/v2

Reply:
agreed